packages/flutter/lib/src/material/button.dart

```diff
@@ -343,9 +355,22 @@ class _RawMaterialButtonState extends State<RawMaterialButton> {
   Widget build(BuildContext context) {
     final Color effectiveTextColor = MaterialStateProperty.resolveAs<Color>(widget.textStyle?.color, _states);
     final ShapeBorder effectiveShape =  MaterialStateProperty.resolveAs<ShapeBorder>(widget.shape, _states);
+    final Offset densityAdjustment = widget.density.adjustment;
+    final BoxConstraints densityConstraints = widget.constraints.copyWith(
+      minWidth: widget.constraints.minWidth != null ? (widget.constraints.minWidth + densityAdjustment.dx).clamp(0.0, double.infinity) : null,
+      minHeight: widget.constraints.minWidth != null ? (widget.constraints.minHeight + densityAdjustment.dy).clamp(0.0, double.infinity) : null,
+    );
+    final EdgeInsetsGeometry padding = widget.padding.add(
```

Comment:
Could be

```dart
    final EdgeInsetsGeometry padding = _clampInsets(
      widget.padding + EdgeInsets.fromLTRB(
        densityAdjustment.dx,
        densityAdjustment.dy,
        densityAdjustment.dx,
        densityAdjustment.dy,
      ),
    );
```
Or (sorry this is a gratuitous suggestion on top of a gratuitous suggestion)

```dart
    final double dy = densityAdjustment.dx;
    final double dy = densityAdjustment.dy;
    final EdgeInsetsGeometry padding = _clampInsets(widget.padding + EdgeInsets.fromLTRB(dx, dy, dx, dy));
```